tests/components/azure_event_hub/conftest.py

```diff
@@ -0,0 +1,126 @@
+from homeassistant.const import STATE_ON
+from homeassistant.helpers.entityfilter import FILTER_SCHEMA
+
+from .const import AZURE_EVENT_HUB_PATH, BASIC_OPTIONS, PRODUCER_PATH, SAS_CONFIG_FULL
+
+from tests.common import MockConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# fixtures for both init and config flow tests
+@pytest.fixture(autouse=True, name="mock_get_eventhub_properties")
+def mock_get_eventhub_properties_fixture():
+    """Mock azure event hub properties, used to test the connection."""
+    with patch(f"{PRODUCER_PATH}.get_eventhub_properties") as get_eventhub_properties:
+        yield get_eventhub_properties
+
+
+@pytest.fixture(name="filter_schema")
+def mock_filter_schema():
+    """Return an empty filter."""
+    yield {}
+
+
+@pytest.fixture(name="entry")
+async def mock_entry_fixture(hass, filter_schema, mock_create_batch):
+    """Create the setup in HA."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    hass.data[DOMAIN] = {DATA_FILTER: FILTER_SCHEMA(filter_schema)}
+    entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(entry.entry_id)
+    assert entry.state == ConfigEntryState.LOADED
+
+    # Clear the component_loaded event from the queue.
+    await hass.data[DOMAIN][DATA_HUB].queue.get()
+    hass.data[DOMAIN][DATA_HUB].queue.task_done()
+
+    yield entry
+
+
+# fixtures for init tests
+@dataclass
+class FilterTest:
+    """Class for capturing a filter test."""
+
+    entity_id: str
+    should_pass: bool
+
+
+@pytest.fixture(name="mock_send_batch", scope="module")
+def mock_send_batch_fixture():
+    """Mock send_batch."""
+    with patch(f"{PRODUCER_PATH}.send_batch") as mock_send_batch:
+        yield mock_send_batch
+
+
+@pytest.fixture(autouse=True, name="mock_client", scope="module")
+def mock_client_fixture(mock_send_batch):
+    """Mock the azure event hub producer client."""
+    with patch(f"{PRODUCER_PATH}.close") as mock_close, patch(
+        f"{PRODUCER_PATH}.__init__", return_value=None
+    ) as mock_init:
+        yield (
+            mock_init,
+            mock_send_batch,
+            mock_close,
+        )
+
+
+@pytest.fixture(name="mock_create_batch")
+def mock_create_batch_fixture():
+    """Mock batch creator and return mocked batch object."""
+    mock_batch = MagicMock()
+    with patch(f"{PRODUCER_PATH}.create_batch", return_value=mock_batch):
+        yield mock_batch
+
+
+@pytest.fixture(name="hub")
+async def mock_hub(hass, entry):
+    """Use the entry and add a single test event to the queue."""
+    assert entry.state == ConfigEntryState.LOADED
+    assert hass.data[DOMAIN][DATA_HUB].queue.qsize() == 0
+
+    hass.states.async_set("sensor.test", STATE_ON)
+    await hass.async_block_till_done()
+
+    assert hass.data[DOMAIN][DATA_HUB].queue.qsize() == 1
+    yield hass.data[DOMAIN][DATA_HUB]
```

Comment:
Don't yield the hub. We shouldn't access this details at all in the tests.

Reply:
so how should I go about testing the internal workings of the hub, create the hub directly in the test_init file, completely skipping over the setup for those tests?

Reply:
We're not interested in asserting the details of the hub. We only want to make sure that the correct library calls are made when the core event bus fires events, and that we handle the interaction with the library correctly.

Reply:
ah, got it now, added in the time_changed function, and that works, the only thing I can't check is the `test_options_update`, if I disable that function, I lose the full coverage (missing the two lines that do the actual update in the update_options function, not sure if that is ok?

Reply:
ok, i think i got them all now, just the `test_late_event` that still uses a patch you might not like, but take a look and let me know if there is a way to do that one!